src/hooks/useArweaveService.ts

```diff
@@ -47,38 +73,26 @@ const useArchaeologistService = () => {
     return tx.id;
   };
 
-  const updateStatus = async () => {
+  const updateStatus = async (arweareTxId: string) => {
     const arweave = initArweave();
-    const currentTextArweaveAddress = 'Xm17-cZJjcx-jc_UL5me1o5nfqC2T1mF-yu03gmKeK4';
-    const txId = await arweave.wallets.getLastTransactionID(currentTextArweaveAddress);
 
-    const res = await arweave.transactions.getStatus(txId);
-    if (res && res.status === 200) {
+    const res = await arweave.transactions.getStatus(arweareTxId);
+    if (res && res.status === 200 && res.confirmed !== null) {
```

Comment:
could `res.confirmed` be undefined here? If not, ignore this comment.

But if it could, would be better to write this as:
`if (res && res.confirmed && res.status === 200)`

Reply:
it can only be null (at least according to the typing)

Reply:
oh, yeah, that is absolutely better.